ts/Core/Series/Series.ts

```diff
@@ -2302,9 +2302,9 @@ class Series {
             // some API data
             point.category = (
                 categories &&
-                typeof (categories as any)[point.x as any] !== 'undefined' ?
-                    (categories as any)[point.x as any] :
-                    point.x
+                typeof categories[point.x] !== 'undefined' ?
+                    categories[point.x] :
+                    point.x as any
```

Comment:
Hmm... I think now it can be further simplified by 'pick': 
```ts
            point.category = (
                categories &&
                pick(categories[point.x], point.x)
            );
```

Reply:
I think the following is what was intended:

```ts 
point.category = pick(categories && categories[point.x as number], point.x);
```
But to make category true to its type (string) you would need:
```ts 
point.category = pick(categories && categories[point.x as number], String(point.x));
```